tests/unit/test_signature/test_parsing.py

```diff
@@ -145,44 +137,27 @@ def test(dep: int, param: int, optional_dep: Optional[int] = Dependency()) -> No
     ) as client:
         response = client.get("/?param=13")
 
-    assert response.json() == {
-        "detail": "Internal Server Error",
-        "extra": error_extra,
-        "status_code": HTTP_500_INTERNAL_SERVER_ERROR,
-    }
+    assert response.json() == {"detail": "Internal Server Error", "status_code": HTTP_500_INTERNAL_SERVER_ERROR}
 
 
-@pytest.mark.parametrize(
-    "preferred_validation_backend, error_extra",
-    (
-        (
-            "attrs",
-            [{"key": "param", "message": "invalid literal for int() with base 10: 'thirteen'", "source": "query"}],
-        ),
-    ),
-)
-def test_validation_failure_raises_400(
-    preferred_validation_backend: Literal["attrs", "pydantic"], error_extra: Any
-) -> None:
+def test_validation_failure_raises_400() -> None:
     dependencies = {"dep": Provide(lambda: 13, sync_to_thread=False)}
 
     @get("/")
     def test(dep: int, param: int, optional_dep: Optional[int] = Dependency()) -> None:
         ...
 
-    with create_test_client(
-        route_handlers=[test], dependencies=dependencies, _preferred_validation_backend=preferred_validation_backend
-    ) as client:
+    with create_test_client(route_handlers=[test], dependencies=dependencies) as client:
         response = client.get("/?param=thirteen")
 
     assert response.json() == {
         "detail": "Validation failed for GET http://testserver.local/?param=thirteen",
-        "extra": error_extra,
+        "extra": [{"key": "param", "message": "Expected `int`, got `str`", "source": "query"}],
```

Comment:
If we parse out expected and actual, we can probably format our own error message, and maybe even include some structured data of the error, e.g., `{"expected": "integer", "received": "string"}`.

Related: https://github.com/jcrist/msgspec/issues/264

Reply:
If you have specific information you'd like to get in an error message, a comment in that issue adding additional context would be most useful.

Reply:
done.